src/database/adapters/mongo.ts

```diff
@@ -42,11 +42,29 @@ export const create = (db: string, collection: string): BidDatabaseService => {
     return result;
   };
 
+  const getBidBySignedMessage = async (
+    signedMessage: string
+  ): Promise<Bid | null> => {
+    const result: Bid | null = await mongo.findOne(database, collection, {
+      signedMessage: `${signedMessage}`,
+    });
+    return result;
+  };
+
+  const cancelBid = async (signedMessage: string): Promise<boolean> => {
```

Comment:
Deleting data out of a database is a very destructive action since there is no way to go back.
Instead of deleting the bid, can we mark a flag on it "canceled"?

This way we still have the bid in the database (for historical or debugging purposes) and we just filter out any bids that have `canceled = true` when returning a list of bids.